Collections_HW/homework2/collections/DoublyLinkedList.java

```diff
@@ -0,0 +1,115 @@
+package homework2.collections;
+
+import java.util.Iterator;
+
+public class DoublyLinkedList implements Iterable<Student> {
+    private Node head;
+    private Node tail;
+    private int size;
+
+    private static class Node {
+        Student student;
+        Node next;
+        Node prev;
+
+        public Node(Student student, Node next, Node prev) {
+            this.student = student;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    // constructor to create a list
+    public DoublyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
+    // inserts the student at the front of the list
+    public void push(Student student) {
+        Node newHead = new Node(student, head, null);
+        if (head != null) {
+            head.prev = newHead;
+        }
+        head = newHead;
+        if (tail == null) {
+            tail = newHead;
+        }
+        size++;
+    }
+
+    // removes and returns the first student of the list
+    public Student pop() {
+        if (size == 0) {
```

Comment:
Shouldn't we call return here?

Reply:
In task's requirments method is: public void push(Student student)

But I can change void to boolean, and it will return true. Would it be better?

Reply:
Sorry, I don't understand what I should change here.

Reply:
If the action is impossible, we should exit the method.

Reply:
Thank you.